importers/data_sources/importer.py

```diff
@@ -0,0 +1,192 @@
+        env_data_dir = self.get_config('ROOT_DATA_PATH', None)
+        if env_data_dir is not None:
+            note('info', '     (Taking data dir from environment variable "RES_ROOT_DATA_PATH")')
+            data_dir = env_data_dir
+        else:
+            note('info', '     (Taking data dir from default)')
+            data_dir = default_data_dir
+        note('info', f'     data_dir: "{data_dir}"')
+
+        # The save_dataset method expects a list of documents
+        # to save, so we are already set!
+        schema_file = os.path.join(get_dataset_schema_dir(), "data_sources_nodes.yaml")
+        validator = get_schema_validator(schema_file=schema_file)
+
+        file_path = os.path.join(data_dir, 'data_sources.json')
+
+        try:
+            with open(file_path, 'r') as data_file:
+                data_sources = json.load(data_file)
+        except OSError as ose:
+            note('error', 'Error loading import data file')
+            return
+
+        data_sources_to_save = []
+        for data_source in data_sources:
+            if not validator.is_valid(data_source):
+                for e in sorted(validator.iter_errors(data_source), key=str):
+                    is_error = True
+                    note('error', f'Validation error: {e.message}')
+                continue
+            else:
+                data_source['_key'] = data_source['ns']
+                data_sources_to_save.append(data_source)
+
+        if is_error:
+            note('error', 'Data did not validate')
+            if dry_run:
+                note('error', 'Dry run completed with errors')
+            else:
+                note('error', 'Due to errors, data will not be loaded')
+            return
+
+        note('success', 'Data loaded and validated successfully')
+
+        # if there are no errors then save the dataset unless this is a dry run
+        if dry_run:
+            note('success', 'Dry run completed successfully')
+            note('warning', 'REMEMBER: Data not loaded')
+        else:
+            self.save_docs('data_sources_nodes', data_sources_to_save)
+
+    def save_docs(self, collection, docs, on_duplicate="update"):
+        """  Saves the source_data docs via into the RE database via the RE api"""
+        resp = requests.put(
+            f'{self.get_config_or_fail("API_URL")}/api/v1/documents',
+            params={
+                "collection": collection,
+                "on_duplicate": on_duplicate
+            },
+            headers={
+                "Authorization": self.get_config_or_fail("AUTH_TOKEN")
+            },
+            data="\n".join(json.dumps(d) for d in docs),
+        )
+        if not resp.ok:
+            raise RuntimeError(resp.text)
+
+        note('success', f"Saved docs to collection {collection}!")
+        for key, value in resp.json().items():
+            note('info', f'     {key}: {value}')
+        return resp
+
+
+def main():
```

Comment:
have you written any tests for module?

Reply:
sorry, do you mean tests for importer.py? Yes, added today.

Reply:
Excellent, thanks -- I will re-review tomorrow.